src/adapter/src/catalog.rs

```diff
@@ -259,6 +259,23 @@ impl CatalogState {
         }
     }
 
+    /// Indicates whether the indicated item is considered stable or not.
+    ///
+    /// Only stable items can be used as dependencies of other catalog items.
+    pub fn is_stable(&self, id: GlobalId) -> bool {
+        let item = self.get_entry(&id).item();
+
+        match item {
+            CatalogItem::Table(_) => {
+                !(id == self.resolve_builtin_table(&MZ_VIEW_KEYS)
+                    || id == self.resolve_builtin_table(&MZ_VIEW_FOREIGN_KEYS))
+            }
+            CatalogItem::Log(_) => false,
+            // In general, an item is stable iff all its dependencies are stable.
+            item => item.uses().iter().all(|id| self.is_stable(*id)),
```

Comment:
I *think* that `uses()` will only return direct parents of an item. So you have something like `unstable system_table -> system_view_1 -> system_view_2`, then `system_view_2` will incorrectly be labelled as stable.

Reply:
Wait, we are calling `is_stable` recursively, so in your example:
* `is_stable(system_view_2)`
  * calls `is_stable(system_view_1)`
    * calls `is_stable(unstable system_table)`
        * returns `true`
    * returns `true`
  * returns `true`

Or am I missing something?

Reply:
Oh no, sorry I misread this. This looks fine to me.